gpkit/tests/t_model.py

```diff
@@ -42,6 +42,16 @@ def test_trivial_gp(self):
                                self.ndig)
         self.assertAlmostEqual(sol["cost"], 2*math.sqrt(2), self.ndig)
 
+    def test_601(self):
+        # tautological monomials should solve but not pass to the solver
+        x = Variable("x")
+        y = Variable("y", 2)
+        m = Model(x,
+                  [x >= 1,
+                   y == 2])
+        m.solve(verbosity=0)
+        self.assertEqual(len(m.program.constraints), 2)
```

Comment:
Oh, the first constraint is the cost
On May 14, 2016 14:56, "Warren Hoburg" <notifications@github.com> wrote:

> In gpkit/tests/t_model.py
> <https://github.com/hoburg/gpkit/pull/719#discussion_r63280418>:
>
> > @@ -42,6 +42,16 @@ def test_trivial_gp(self):
> >                                 self.ndig)
> >          self.assertAlmostEqual(sol["cost"], 2*math.sqrt(2), self.ndig)
> >
> > +    def test_601(self):
> > +        # tautological monomials should solve but not pass to the solver
> > +        x = Variable("x")
> > +        y = Variable("y", 2)
> > +        m = Model(x,
> > +                  [x >= 1,
> > +                   y == 2])
> > +        m.solve(verbosity=0)
> > +        self.assertEqual(len(m.program.constraints), 2)
>
> the length 2 assertion seems to conflict with the inline comment above --
> if "tautological monomials should solve but not pass to the solver", then
> shouldn't the length of m.program.constraints be 1?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/hoburg/gpkit/pull/719/files/c243099664b2e4ba255fd0ef353d7f05bc39636a#r63280418>
>